iroh/src/p2p.rs

```diff
@@ -88,3 +95,12 @@ observed addresses: {:?}
         l.peer_id, l.listen_addrs, l.protocols, l.protocol_version, l.observed_addrs
     );
 }
+
+fn display_peers(peers: HashMap<PeerId, Vec<Multiaddr>>) {
+    // let mut pid_str: String;
+    for (peer_id, addrs) in peers {
+        if let Some(addr) = addrs.first() {
```

Comment:
This is just grabbing the first address, which I'm doubtful is actually the address we're connected to. We should make sure the addresses listed here are actually connected

Reply:
We need a rule of thumb for this in `lookup` as well: some entries return 20+ addrs, and that's... not necessary.

For the lookup of the currect peer, we can be more comfortable just displaying the first 5 (or whatever number we choose), because according to the docs they rank the `external_addrs` that we can be dialed at, but like you mention, we don't have that luxury here, or for the other `lookup` commands.

But also, we may want to pair this down. Maybe you use `p2p peers` to just get a list of the peer ids (not peer_ids and addrs), and use `p2p lookup PEER_ID` to investigate possible addresses?